tests/address_test.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+import easypost
+import tests.fixture as fixture
+
+
+@pytest.mark.vcr()
+def test_address_create():
+    address = easypost.Address.create(**fixture.basic_address())
+
+    assert isinstance(address, easypost.Address)
+    assert str.startswith(address.id, "adr_")
+    assert address.street1 == "388 Townsend St"
+
+
+@pytest.mark.vcr()
+def test_address_create_verify_strict():
+    address_data = fixture.basic_address()
+    address_data["verify_strict"] = [True]
+
+    address = easypost.Address.create(**address_data)
+
+    assert isinstance(address, easypost.Address)
+    assert str.startswith(address.id, "adr_")
+    assert address.street1 == "388 TOWNSEND ST APT 20"
+
+
+@pytest.mark.vcr()
+def test_address_retrieve():
+    address = easypost.Address.create(**fixture.basic_address())
+
+    retrieved_address = easypost.Address.retrieve(address.id)
+
+    assert isinstance(retrieved_address, easypost.Address)
+    assert retrieved_address.id == address.id
```

Comment:
# reviewer note:

We can't compare instances here because we have to implement the method [__eq__](https://docs.python.org/3/reference/datamodel.html#object.__eq__) in each class

Reply:
We could probably do some kind of "loop over every property of Object A, get the corresponding property on Object B, compare", but I think that would introduce some funky stuff. This works for now.

Reply:
I played briefly with this. Only comparing IDs isn't ideal when we can reliably compare the entire object (various bugs exposed themselves due to comparing whole objects in other libs). I also implemented `__eq__` in a class to try and didn't have luck, though I may have done so incorrectly on my first attempt.

My suggestion is we try to compare whole objects. I'd suggest doing a bit of extra digging here to see what our options are and give it a try on a couple objects.

Reply:
I have created a helper function that is going to compare each attribute recursively. We do have to convert the instance to dictionary first for the comparsion.

Reply:
My own customized helper function didn't work well because the response object can have different types of attributes. Spent a few hours trying to make my helper functions work but later realized we can use [deepdiff](https://pypi.org/project/deepdiff/) to compare two instances if that's okay?

Reply:
I'd rather not introduce another dependency if we can get away without it. Thinking on this a bit, I believe I may have seen an implementation of something like this from @jontsai in the past (that we use internally) that may not even be that many lines of code that we could snag and add here. We can ask him about this.